pyproject.toml

```diff
@@ -0,0 +1,28 @@
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "cps"
+version = "0.dev"
+description = ""
+authors = ["The CPS Project <@>"]
+readme = "README.md"
+packages = [
+  { include = "cps.py", from = "_extensions" },
+  { include = "autosectionlabel.py", from = "_extensions" }
+]
+
+[tool.poetry.dependencies]
+python = "^3.10"
```

Comment:
BTW, if I only have Python 3.12 locally and try to run `poetry`, does it just use 3.12? Or does it go obtain a local copy of 3.10? (For that matter, what does the `^` mean?)

Reply:
For information on the version specifications, see poetry's documentation on [dependency specifications](https://python-poetry.org/docs/dependency-specification/)

To succinctly answer your question, `^` means "any newer minor version, but not a newer major version". So as long as Python 4 isn't a requirement we'll be fine 😄 

And yes it'll use your local python. If you want to use different versions of python via poetry, you can install multiple versions of poetry with `pipx` using a different version for each:

```console
$ pipx install poetry --python 3.10 --suffix @3.10
$ poetry@3.10 install --with=docs
$ poetry@3.10 run ...
```